fdbclient/include/fdbclient/CommitTransaction.h

```diff
@@ -111,23 +117,93 @@ struct MutationRef {
 	}
 
 	std::string toString() const {
-		return format("code: %s param1: %s param2: %s",
+		std::string checksumStr;
+		if (checksum.present()) {
+			checksumStr = format("checksum: %s ", std::to_string(checksum.get()).c_str());
+		}
+		return format("%scode: %s param1: %s param2: %s",
 		              type < MutationRef::MAX_ATOMIC_OP ? typeString[(int)type] : "Unset",
+		              checksumStr.c_str(),
 		              printable(param1).c_str(),
 		              printable(param2).c_str());
 	}
 
+	uint8_t typeWithChecksum() const { return this->type | CHECKSUM_FLAG_MASK; }
+	void removeChecksum() {
```

Comment:
Do we want to validate checksum when removing the checksum? Otherwise, we are not able to know the correctness of the current mutation when using it (e.g. applyMetadataMutation).

Reply:
In ApplyMetadataMutation, this is called after a mutation is copied from an existing one, and this will make sure any following modification will start with a clean state.